src/Validator.sol

```diff
@@ -0,0 +1,212 @@
+pragma solidity ^0.8.19;
+
+import {IProposalTypesConfigurator} from "src/interfaces/IProposalTypesConfigurator.sol";
+import {IAgoraGovernor} from "src/interfaces/IAgoraGovernor.sol";
+
+/**
+ * Contract that stores proposalTypes for the Agora Governor.
+ */
+contract Validator {
+    error InvalidParamNotEqual();
+    error InvalidParamRange();
+
+    function validate_uint8(
+        bytes calldata param,
+        bytes calldata scopedParam,
+        IProposalTypesConfigurator.Comparators comparison
+    ) public pure {
+        uint8 typedParam = uint8(bytes1(param[0:1]));
+        uint8 scopedParamTyped = uint8(bytes1(scopedParam[0:1]));
+
+        if (comparison == IProposalTypesConfigurator.Comparators.EQUAL) {
+            if (scopedParamTyped != typedParam) revert InvalidParamNotEqual();
+        }
+
+        if (comparison == IProposalTypesConfigurator.Comparators.LESS_THAN) {
+            if (typedParam >= scopedParamTyped) {
+                revert InvalidParamRange();
+            }
+        }
+
+        if (comparison == IProposalTypesConfigurator.Comparators.GREATER_THAN) {
+            if (typedParam <= scopedParamTyped) {
+                revert InvalidParamRange();
+            }
+        }
```

Comment:
we can make this an internal function instead of duplicating it for all functions

Reply:
Ideally yes, but after the type conversion was unsure how to do so since each one would have different types for the parameters no?

Reply:
`bytes32` is a superset of everything so maybe we can just have every type cast to bytes32 and the comparision checks are done on bytes32, which should equivalent behavior as in uint for example

Reply:
so each `validate` function takes in the given parameter (say uint32), casts to bytes32, and passes to the internal function which does the comparision check on the bytes32 variables

Reply:
Ah I see, I guess after we do this check with the sizes, things are now padded correctly and we can then perform the `bytes32` conversion